packages/lodestar/test/sim/merge-interop.test.ts

```diff
@@ -58,135 +60,75 @@ describe("executionEngine / ExecutionEngineHttp", function () {
-    // Note: Use double quotes on paths to let bash expand the ~ character
-    await shell(`${process.env.GETH_BINARY_PATH} --catalyst --datadir "${dataPath}" init "${genesisPath}"`);
-
-    startGethProcess(["--catalyst", "--http", "--ws", "-http.api", "engine,net,eth", "--datadir", dataPath]);
-
-    // Wait for Geth to be online
-    const controller = new AbortController();
-    afterEachCallbacks.push(() => controller?.abort());
-    await waitForGethOnline(jsonRpcUrl, controller.signal);
-
-    // Fetch genesis block hash
-    const genesisBlockHash = await getGenesisBlockHash(jsonRpcUrl, controller.signal);
-    return {genesisBlockHash};
-  }
-
-  // Ref: https://notes.ethereum.org/_UH57VUPRrC-re3ubtmo2w
-  // Build geth from source at branch https://github.com/ethereum/go-ethereum/pull/23607
-  async function runGethPreMerge(): Promise<{genesisBlockHash: string}> {
-    const privKey = "45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8";
-    const pubKey = "0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b";
-    const password = "12345678";
-
-    if (!process.env.GETH_BINARY_PATH) {
-      throw Error("GETH_BINARY_PATH ENV must be provided");
+  async function runEL(elScript: string, ttd: number): Promise<{genesisBlockHash: string}> {
+    if (!process.env.EL_BINARY_PATH || !process.env.EL_DIR || !process.env.EL_PORT) {
+      throw Error(
+        `EL ENV must be provided, EL_BINARY_PATH: ${process.env.EL_BINARY_PATH}, EL_DIR: ${process.env.EL_DIR}, EL_PORT: ${process.env.EL_PORT}`
+      );
     }
 
     await shell(`rm -rf ${dataPath}`);
     fs.mkdirSync(dataPath, {recursive: true});
 
-    const genesisPath = path.join(dataPath, "genesis.json");
-    const skPath = path.join(dataPath, "sk.json");
-    const passwordPath = path.join(dataPath, "password.txt");
-    fs.writeFileSync(genesisPath, JSON.stringify(genesisGethPreMerge, null, 2));
-    fs.writeFileSync(skPath, privKey);
-    fs.writeFileSync(passwordPath, password);
-
-    // Use as check to ensure geth is available and built correctly.
-    // Note: Use double quotes on paths to let bash expand the ~ character
-    //  ./build/bin/geth --catalyst --datadir "~/ethereum/taunus" init genesis.json
-    console.log("Initilizing Geth");
-    await shell(`${process.env.GETH_BINARY_PATH} --catalyst --datadir "${dataPath}" init "${genesisPath}"`);
-
-    // Import the signing key (press enter twice for empty password):
-    //  ./build/bin/geth --catalyst --datadir "~/ethereum/taunus" account import sk.json
-    console.log("Importing account to Geth");
-    await shell(
-      `${process.env.GETH_BINARY_PATH} --catalyst --datadir "${dataPath}" account import "${skPath}" --password ${passwordPath}`
-    );
-
-    startGethProcess([
-      "--catalyst",
-      "--http",
-      "--ws",
-      "-http.api",
-      "engine,net,eth,miner",
-      "--datadir",
+    startELProcess(`../../kintsugi/${process.env.EL_DIR}/${elScript}`, [
+      "--ttd",
+      `${ttd}`,
```

Comment:
`kintsugi/parse-args.sh` is really cool, but could the same be achieved with ENV variables? It's subjective but I believe it makes it easier to follow how the arguments are passed all the way down to the geth and nethermind binaries

Then it would allow to type at the run function level
```ts
function startELProcess(args: {
  runScriptPath: string;
  TTD: string;
  DATA_DIR: string;
  EL_BINARY_PATH: string;
}): void {
```
Then in the spawn do
```ts
spawn(elPath, [], {
  env: {
    ...process.env,
    TTD,
    DATA_DIR,
    EL_BINARY_PATH
  }
})
```

Reply:
yea sounds good! :+1: